### PR TITLE
Allow secrets in token of GitHubPullrequestPoller

### LINKED ISSUE #4921: GitHubPullrequestPoller token secret gives builtins.TypeError:
When attempting the following in GitHubPullrequestPoller:
```
token=util.Secret("githubToken"),
```
I receive: builtins.TypeError: must be str, not Secret

and similarly via Interpolate:
```
token=util.Interpolate("%(secret:githubToken)s"),
```
builtins.TypeError: must be str, not Interpolate

Apologies if I'm doing this wrong but given:
https://docs.buildbot.net/latest/manual/secretsmanagement.html#as-argument-to-services
I had expected this to work. I got the impression that the secret manager was as yet not supported across the board and so if this is an example of an unsupported use then it seems like a nice one to have.

Thanks all for a great project.

Incidentally in both cases I can cast via str() and avoid this issue but then run into another issue which I will report separately.

### PR DESCRIPTION
**Pull request by alexreinking**
What it says in the title. I haven't updated the unit tests because I'm not very familiar with the codebase yet. I have some time to try on my own, but would appreciate any pointers.

Fixes #4921

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by p12tic** [comment]
@alexreinking : Thanks a lot, this looks good.

As for tests, the tests for this class live in master/buildbot/test/unit/changes/test_github.py. Secrets provider can be setup like this (from master/buildbot/test/unit/www/test_oauth.py):

```
        fake_storage_service = FakeSecretStorage()
        fake_storage_service.reconfigService(secretdict={"client-id": "secretClientId",
                                                         "client-secret": "secretClientSecret"})
        secret_service = SecretManager()
        secret_service.services = [fake_storage_service]
        yield secret_service.setServiceParent(self._master)
```

Thanks!

File: master/buildbot/newsfragments/githubpullrequestpoller-secrets.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+``GitHubPullrequestPoller`` now supports secrets in its ``token`` argument
```
Comment (p12tic) [inline master/buildbot/newsfragments/githubpullrequestpoller-secrets.bugfix:1]: Nitpick: append ```(:issue:`4921`) ``` to mention the bug number.

Comment (alexreinking) [reply master/buildbot/newsfragments/githubpullrequestpoller-secrets.bugfix:1]: Done!

**Comment by alexreinking** [comment]
@p12tic - that snippet didn't work for me unless I moved the `reconfigService` call to the end. Looks like both `secret_service.setServiceParent` and `self.master.startService` try to configure the `FakeSecretStorage` service and wind up overwriting the secretdict.

Either way, I added a test by running the same steps as in SimplePR, but with the token set to `Secret('token')`. Let me know if I need to do anything else! I also squashed this down to one commit since my fiddling isn't interesting history-wise.

**Review by p12tic — APPROVED** [review]
Looks great, thanks!

**Comment by alexreinking** [comment]
You're welcome! Thank you for helping me with, reviewing, and ultimately accepting my PR!

**Pull request merged**